leaf/extensions/tags.py

```diff
@@ -0,0 +1,351 @@
+        )
+        embeds = []
+
+        if not tags:
+            # TODO: Remove pagination buttons if there are no tags in the server.
+            embeds.append(
+                discord.Embed(
+                    description="There are no tags in this server.",
+                    color=discord.Color.dark_embed(),
+                )
+            )
+        else:
+            chunks = list(discord.utils.as_chunks(tags, 15))
+            for index, chunk in enumerate(chunks):
+                embed = discord.Embed(
+                    description="\n".join(
+                        [f"• **{tag['name']}** (Uses: {tag['uses']})" for tag in chunk]
+                    ),
+                    color=discord.Color.dark_embed(),
+                )
+                embed.set_footer(text=f"Page {index + 1} / {len(chunks)}")
+                embeds.append(embed)
+
+        if not 0 <= (starting_page - 1) <= len(embeds):
+            await interaction.response.send_message(
+                embed=discord.Embed(
+                    description="That page does not exist.",
+                    color=discord.Color.dark_embed(),
+                ),
+                ephemeral=True,
+            )
+            return
+
+        paginator = Paginator(embeds=embeds, index=starting_page - 1)
+        await paginator.start(interaction, ephemeral=silent)
+
+    @app_commands.describe(tag="The name of the tag to view.")
+    @app_commands.command(name="view", description="Sends the content of a tag.")
+    async def view_tag(
+        self, interaction: discord.Interaction, tag: str, silent: Optional[bool] = False
+    ) -> None:
+        tag_record = await self.bot.database.fetchrow(
+            "SELECT * FROM tags WHERE name = $1 AND guild_id = $2 AND deleted = FALSE",
+            tag,
+            interaction.guild.id,
+        )
+
+        if tag_record:
+            await interaction.response.send_message(
+                embed=discord.Embed(
+                    title=tag_record["name"],
+                    description=tag_record["content"],
+                    color=discord.Color.dark_embed(),
+                ),
+                ephemeral=silent,
+            )
+            await self.bot.database.execute(
+                "UPDATE tags SET uses = $1 WHERE name = $2 and guild_id = $3;",
+                tag_record["uses"] + 1,
+                tag,
+                interaction.guild.id,
+            )
+        else:
+            await interaction.response.send_message(
+                embed=discord.Embed(
+                    description="That tag does not exist.",
+                    color=discord.Color.dark_embed(),
+                ),
+                ephemeral=silent,
+            )
+
+    @app_commands.describe(name="The tag of the newly created tag.")
+    @app_commands.command(name="create", description="Creates a new tag.")
+    async def create_tag(self, interaction: discord.Interaction, name: str) -> None:
```

Comment:
Consider using transactions here and elsewhere where you might be modifying multiple tables or executing multiple queries.